src/ert/_c_wrappers/job_queue/forward_model.py

```diff
@@ -97,10 +97,8 @@ def filter_env_dict(job, d):
                     # replaced by substitute, but were not.
                     result[new_key] = new_value
                 else:
-                    logging.warning(
-                        "Environment variable {} skipped due to unmatched define {}",
-                        new_key,
-                        new_value,
+                    logger.warning(
```

Comment:
The problem is that `{}` should be `%s`. The values need to be given as arguments to the logger to take advantage of the logging lazy message string evaluation: https://stackoverflow.com/questions/4148790/lazy-logger-message-string-evaluation

Reply:
I am guessing that the reason why this wasn't captured by the tests was that there was no logger installed. Could you add testing of the logger output via the `caplog` fixture to `test_that_values_with_brackets_are_ommitted`?

Reply:
Should we really make a test verifying that the input to the log statement is correct?

```
>>> logging.warning("some %s", "some")
WARNING:root:some some
>>> logging.warning("some {}", "some")
--- Logging error ---
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/logging/__init__.py", line 1083, in emit
    msg = self.format(record)
  File "/usr/local/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/logging/__init__.py", line 927, in format
    return fmt.format(record)
  File "/usr/local/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/logging/__init__.py", line 663, in format
    record.message = record.getMessage()
  File "/usr/local/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/logging/__init__.py", line 367, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
```

I don't really understand `test_that_values_with_brackets_are_ommitted` either to be honest. Perhaps there should be a test that enters the `else` statement, as that is clearly an untested code path.

Reply:
The `else` should be entered by `test_that_values_with_brackets_are_ommitted` . I think the problem is that I have made the code hard to read. There is a not in front of the `elif` case. Perhaps it could be a good idea to refactor a bit?